### PR TITLE
Fix breaking changes introduced by newest opentelemetry update

### LINKED ISSUE #151: Package broken with newest opentelemetry dependencies
When using the latest opentelementy packages (0.27.0), the Nest application fails to start with the following error:

```
/home/node/app/node_modules/nestjs-otel/src/metrics/metric-data.ts:27
      const valueRecorder = meter.createValueRecorder(name, options);
                                  ^
TypeError: meter.createValueRecorder is not a function
    at getOrCreateValueRecorder (/home/node/app/node_modules/nestjs-otel/src/metrics/metric-data.ts:27:35)
    at MetricService.getOrCreateValueRecorder (/home/node/app/node_modules/nestjs-otel/src/metrics/metric.service.ts:26:36)
    at MetricService.getValueRecorder (/home/node/app/node_modules/nestjs-otel/src/metrics/metric.service.ts:20:17)
    at new ApiMetricsMiddleware (/home/node/app/node_modules/nestjs-otel/src/middleware/api-metrics.middleware.ts:90:47)
    at Injector.instantiateClass (/home/node/app/node_modules/@nestjs/core/injector/injector.js:301:19)
    at callback (/home/node/app/node_modules/@nestjs/core/injector/injector.js:48:41)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Injector.resolveConstructorParams (/home/node/app/node_modules/@nestjs/core/injector/injector.js:124:24)
    at async Injector.loadInstance (/home/node/app/node_modules/@nestjs/core/injector/injector.js:52:9)
    at async Injector.loadMiddleware (/home/node/app/node_modules/@nestjs/core/injector/injector.js:61:9)
```

### PR DESCRIPTION
**Pull request by adrianmxb**
This PR closes #151 .
I followed through the Upgrade guidelines of opentelemetry-js, some things are looking a bit different.
Should be a good starting point though.

I had to change the spec for the api-metrics middleware test, the .clear() function seems to be missing in the new otel api-metrics spec.

### CODE REVIEW

File: tests/e2e/metrics/metric.service.spec.ts [hunk 1]
```diff
@@ -184,8 +194,8 @@ describe('MetricService', () => {
       // Starts empty
       expect(meterData.size).toBe(0);
 
-      const counter = metricService.getValueRecorder('test1');
-      counter.clear();
+      const counter = metricService.getHistogram('test1');
+      // counter.clear();
```
Comment (adrianmxb) [inline tests/e2e/metrics/metric.service.spec.ts:198]: Clear doesn't exist anymore on histogram

File: tests/e2e/metrics/metric.service.spec.ts [hunk 2]
```diff
@@ -194,24 +204,26 @@ describe('MetricService', () => {
...
 
       app = moduleRef.createNestApplication();
       await app.init();
 
       metricService = moduleRef.get<MetricService>(MetricService);
 
-      const counter = metricService.getValueRecorder('test1', { description: 'test1 description' });
-      counter.clear();
+      const counter = metricService.getHistogram('test1', { description: 'test1 description' });
+      // counter.clear();
```
Comment (adrianmxb) [inline tests/e2e/metrics/metric.service.spec.ts:224]: Clear doesn't exist anymore on histogram

**Review by adrianmxb — COMMENTED** [review]
See comment in PR about missing clear method.

File: src/metrics/decorators/common.ts [hunk 3]
```diff
@@ -1,5 +1,9 @@
 import { Counter, MetricOptions } from '@opentelemetry/api-metrics';
-import { getOrCreateCounter, getOrCreateValueRecorder, MetricType } from '../metric-data';
+import { getOrCreateCounter, MetricType } from '../metric-data';
+
+interface WhatIsThisShit {
```
Comment (pragmaticivan) [inline src/metrics/decorators/common.ts:4]: 😂

Comment (adrianmxb) [reply src/metrics/decorators/common.ts]: lmao, and thats what happens when you get frustrated about your IDEs code formatter. 😅
My bad dude, its gone now.

**Comment by adrianmxb** [comment]
Converted this to a DRAFT for now, will check if everything works out later.
Was a bit rushed yesterday night.

**Comment by pragmaticivan** [comment]
ping @adrianmxb Do you need any help with this PR?

**Comment by adrianmxb** [comment]
@pragmaticivan really stressful week, will look into it hopefully beginning of next week. Got some issues with metrics.

**Comment by adrianmxb** [comment]
@pragmaticivan The dependency on opentelemetry-node-metrics seems to be a problem.
The library hasn't adopted the latest opentelemetry-js version yet which breaks host and default metrics.

**Comment by pragmaticivan** [comment]
Hi @adrianmxb I've pushed some changes in the mainstream. I added you as co-author here https://github.com/pragmaticivan/nestjs-otel/commit/6ed63aefbc1e39da9659d6e232a4883f7ed93dac

Will only release once https://www.npmjs.com/package/@opentelemetry/host-metrics gets the latest update to 0.27.0

**Comment by adrianmxb** [comment]
@pragmaticivan did you see the referenced issue i linked in #151 ?
You should check that out before doing a new release.

**Comment by pragmaticivan** [comment]
Hi @adrianmxb I'm actually about to drop this library you commented in order to use https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/packages/opentelemetry-host-metrics instead. Which is maintained by open telemetry js team.

**Pull request closed without merging**